fw/http_frame.c

```diff
@@ -394,30 +394,50 @@ tfw_h2_send_wnd_update(TfwH2Ctx *ctx, unsigned int id, unsigned int wnd_incr)
 static inline int
 tfw_h2_send_settings_init(TfwH2Ctx *ctx)
 {
-	unsigned char key_buf[SETTINGS_KEY_SIZE];
-	unsigned char val_buf[SETTINGS_VAL_SIZE];
+	/* XX(id, val) where
+	 * `id` is one of HTTP2_SETTINGS_xxx constants
+	 * `val` is of the size uint32_t
+	 */
+#define INITIAL_HEADERS(XX)                                     \
+	XX(HTTP2_SETTINGS_TABLE_SIZE, HPACK_ENC_TABLE_MAX_SIZE) \
+	XX(HTTP2_SETTINGS_INIT_WND_SIZE, ctx->lsettings.wnd_sz)
+
+#define PLUS_ONE(...) + 1
+	/* Automatically adjusted payload buffer */
+	unsigned char payload[(SETTINGS_KEY_SIZE + SETTINGS_VAL_SIZE)
+		* (0 INITIAL_HEADERS(PLUS_ONE))];
```

Comment:
The macro usage is fancy, but makes the code harder to read and doesn't reduce copy & paste since the most macros are used only once and only one of them twice.